api/src/main/java/org/openmrs/api/impl/GlobalLocaleList.java

```diff
@@ -17,58 +17,59 @@
 import org.openmrs.api.GlobalPropertyListener;
 import org.openmrs.util.LocaleUtility;
 import org.openmrs.util.OpenmrsConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
- * A utility class which caches the current list of allowed locales, rebuilding the list whenever
- * the global properties are updated.
+ * A utility class which caches the current list of allowed locales, rebuilding
+ * the list whenever the global properties are updated.
  */
 public class GlobalLocaleList implements GlobalPropertyListener {
 	
+	protected final Logger log = LoggerFactory.getLogger(getClass());
+
 	private Set<Locale> allowedLocales = null;
-	
+
 	/**
 	 * @see org.openmrs.api.GlobalPropertyListener#globalPropertyChanged(org.openmrs.GlobalProperty)
 	 */
 	@Override
 	public void globalPropertyChanged(GlobalProperty newValue) {
 		allowedLocales = new LinkedHashSet<Locale>();
 		for (String allowedLocaleString : newValue.getPropertyValue().split(",")) {
-			try {
-				Locale allowedLocale = LocaleUtility.fromSpecification(allowedLocaleString.trim());
-				if (allowedLocale != null) {
-					allowedLocales.add(allowedLocale);
-				}
-			}
-			catch (Exception e) {
-				// bad locale spec? just ignore it. the UI should take care of
-				// guiding the user.
+			Locale allowedLocale = LocaleUtility.fromSpecification(allowedLocaleString.trim());
+			if (allowedLocale != null) {
+				allowedLocales.add(allowedLocale);
+			} else {
+				log.error("Attempt to set invalid locale '{}'", allowedLocaleString);
```

Comment:
Why is this change needed? I thought you just have to remove the try...catch clause and no other change